src/apdu.c

```diff
@@ -33,6 +33,26 @@
 #include <stdint.h>
 #include <string.h>
 
+tz_exc read_path_with_curve(derivation_type_t derivation_type,
```

Comment:
The derivation type is selected at the time of setup. It can not be changed during signing. so it makes sense to derivce the public key when read_path_with_curve is called from handle_setup. Rather create a new function setup_path_with_curve and derive public key there along with a call to read_path_with_curve. Call setup_path_with_curve from handle_setup and read_path_with_curve from other places.

Reply:
The `SETUP` instruction is not required to sign:
 - `AUTHORIZE_BAKING` will setup the authorized key too.
 - If the authorized key was present in the RAM, `SIGN` with index 0x00 will tell which public key to sign with

The aims of this function is too make sure that, every time the stored Bip32-path-with-curve is updated, the stored PK is updated to.